proto/regen/data/v1alpha2/query.proto

```diff
@@ -12,8 +12,8 @@ option go_package = "github.com/regen-network/regen-ledger/x/data";
 // Query is the regen.data.v1alpha2 Query service
 service Query {
   // ByHash queries data based on its ContentHash.
-  rpc ByHash (QueryByHashRequest) returns (QueryByHashResponse) {
-    option (google.api.http).get = "/regen/data/v1alpha2/content/by_hash";
+  rpc ByIRI (QueryByIRIRequest) returns (QueryByIRIResponse) {
+    option (google.api.http).get = "/regen/data/v1alpha2/content/{iri}";
```

Comment:
after a call with @clevinson we thought querying by iri makes more sense here. 

thoughts on this design? @aaronc @ryanchristo

Reply:
Makes sense to me. More aligned with the signers query.

Reply:
Sure but ideally we have both. Or maybe we add query methods to convert between IRIs and hashes but always query with IRIs by default?